src/components/accounts/SetRecoveryInfo.js

```diff
@@ -0,0 +1,88 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { isValidPhoneNumber } from 'react-phone-number-input'
+
+import { Wallet } from '../../utils/wallet'
+
+import SetRecoveryInfoSection from './SetRecoveryInfoSection'
+import SetRecoveryInfoForm from './SetRecoveryInfoForm'
+import SetRecoveryInfoContainer from './SetRecoveryInfoContainer'
+import { requestCode, validateCode } from '../../actions/account';
+
+class SetRecoveryInfo extends Component {
+   state = {
+      loader: false,
+      phoneNumber: '',
+      isLegit: false,
+   }
+
+   componentDidMount = () => {
+      this.wallet = new Wallet()
+   }
+
+   handleChange = (e, { name, value }) => {
+      this.setState(() => ({
+         [name]: value,
+         isLegit: this.isLegitField(name, value)
+      }))
+   }
+
+   isLegitField(name, value) {
```

Comment:
let's rename legit to something else, maybe valid?

Reply:
this seems subjective, `Legit` is used in all other code in wallet.